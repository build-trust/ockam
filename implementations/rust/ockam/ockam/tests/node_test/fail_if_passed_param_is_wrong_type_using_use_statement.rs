// This test checks that #[ockam_test_macros::node_test] causes a compile time error
// if the function is passed a param that is not of type `ockam_node::Context`
// The param is not fully qualified (ie. using `use` statement).

#[ockam_test_macros::node_test]
async fn my_test(ctx: String) -> ockam_core::Result<()> {}
