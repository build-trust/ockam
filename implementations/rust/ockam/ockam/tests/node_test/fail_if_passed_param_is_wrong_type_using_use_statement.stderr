error: Expected `ockam::Context` found `String`
 --> tests/node_test/fail_if_passed_param_is_wrong_type_using_use_statement.rs:6:23
  |
6 | async fn my_test(ctx: String) -> ockam_core::Result<()> {}
  |                       ^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/node_test/fail_if_passed_param_is_wrong_type_using_use_statement.rs:5:1
  |
5 | / #[ockam_test_macros::node_test]
6 | | async fn my_test(ctx: String) -> ockam_core::Result<()> {}
  | |__________________________________________________________^ consider adding a `main` function to `$DIR/tests/node_test/fail_if_passed_param_is_wrong_type_using_use_statement.rs`
